http.c

```diff
@@ -2489,7 +2533,20 @@ evhttp_connection_connect_(struct evhttp_connection *evcon)
 			socklen = sizeof(struct sockaddr_in6);
 		}
 		ret = bufferevent_socket_connect(evcon->bufev, sa, socklen);
-	} else {
+	}
+#ifndef _WIN32
+	else if (evcon->unixsocket) {
+		struct sockaddr_un sockaddr;
+		sockaddr.sun_family = AF_UNIX;
+		if (strlen(evcon->unixsocket) >= sizeof(sockaddr.sun_path)) {
```

Comment:
this check should be in `evhttp_connection_unixsocket_new`

Reply:
done